lib/model/Transaction.js

```diff
@@ -48,11 +50,13 @@ Transaction.prototype.complete = function(callback) {
 
   var path = 'accounts/' + self.account.id +  '/transactions/' + self.id + '/complete';
 
-  self.client._postHttp(path, null, function onPut(err, result) {
-    if (!handleError(err, result, callback)) {
-      callback(null, new Transaction(self.client, result.data, self.account));
-    }
-  });
+  return this._createPromise(function (resolve, reject) {
+    self.client._postHttp(path, null, function onPut(err, result) {
+      if (!handleError(err, result, reject)) {
+        resolve([new Transaction(self.client, result.data, self.account)]);
```

Comment:
You seem to always be returning an Array in the resolve, which makes working with the Promise a bit difficult in my opinion. Why not just return the actual object? In this case, why not just return new Transaction(...)

Since you're returning arrays, the caller must always get the array and then inspect index 0 of it. 

In the async/await world, here's what I would want:

async createNewTransaction() {
     let transaction = await client.createTransaction(...);
}

With the array, I have to do:

     let transactionResult = await client.createTransaction(...);
     let transaction = transactionResult[0];

Reply:
As is wrote I decided to return the result as array to support pagination on the one hand and to be consistent between all functions.